qiskit/compiler/transpiler.py

```diff
@@ -335,73 +356,92 @@ def callback_func(**kwargs):
+            # Only parse backend properties when the target isn't skipped to
+            # preserve the former behavior of transpile.
+            backend_properties = _parse_backend_properties(backend_properties, backend)
+            # Build target from constraints.
+            target = Target.from_configuration(
+                basis_gates=basis_gates,
+                num_qubits=backend.num_qubits if backend is not None else None,
+                coupling_map=coupling_map,
+                # If the instruction map has custom gates, do not give as config, the information
+                # will be added to the target with update_from_instruction_schedule_map
+                inst_map=inst_map if inst_map and not inst_map.has_custom_gate() else None,
+                backend_properties=backend_properties,
+                instruction_durations=instruction_durations,
+                concurrent_measurements=(
+                    backend.target.concurrent_measurements if backend is not None else None
+                ),
+                dt=dt,
+                timing_constraints=timing_constraints,
+                custom_name_mapping=name_mapping,
+            )
+
+    # Update target with custom gate information. Note that this is an exception to the priority
+    # order (target > loose constraints), added to handle custom gates for scheduling passes.
+    if target is not None and _given_inst_map and inst_map.has_custom_gate():
         target = copy.deepcopy(target)
         target.update_from_instruction_schedule_map(inst_map)
 
-    if not ignore_backend_supplied_default_methods:
-        if scheduling_method is None and hasattr(backend, "get_scheduling_stage_plugin"):
-            scheduling_method = backend.get_scheduling_stage_plugin()
-        if translation_method is None and hasattr(backend, "get_translation_stage_plugin"):
-            translation_method = backend.get_translation_stage_plugin()
-
+    # Edge cases require using the old model (loose constraints) instead of building a target,
+    # byt we don't populate the passmanager config with loose constraints unless it's one of
+    # the known edge cases to control the execution path.
     pm = generate_preset_pass_manager(
         optimization_level,
-        backend=backend,
         target=target,
-        basis_gates=basis_gates,
-        inst_map=inst_map,
-        coupling_map=coupling_map,
-        instruction_durations=instruction_durations,
-        backend_properties=backend_properties,
-        timing_constraints=timing_constraints,
+        basis_gates=basis_gates if _skip_target else None,
+        coupling_map=coupling_map if _skip_target else None,
+        instruction_durations=instruction_durations if _skip_target else None,
+        backend_properties=backend_properties if _skip_target else None,
+        timing_constraints=timing_constraints if _skip_target else None,
+        inst_map=inst_map if _skip_target else None,
```

Comment:
I feel like this would be simpler if we moved this logic down into `generate_preset_pass_manager()` itself. We already expose skip_target there so internally we can change that function to generate the target instead of doing it in transpile. (it might also require moving #11996 to `generate_preset_pass_manager()` too).